Marlin/src/HAL/HAL_STM32/HAL.cpp

```diff
@@ -78,14 +78,20 @@ uint16_t HAL_adc_result;
 // Public functions
 // --------------------------------------------------------------------------
 
-// HAL initialization task
-void HAL_init(void) {
 
-  // Needed for DELAY_NS() / DELAY_US() on CORTEX-M7
-  #if (defined(__arm__) || defined(__thumb__)) && __CORTEX_M == 7
+// Needed for DELAY_NS() / DELAY_US() on CORTEX-M7
+#if (defined(__arm__) || defined(__thumb__)) && __CORTEX_M == 7
+  // HAL pre-initialization task
+  // Force the preinit function to run between the premain() and main() function
+  // of the STM32 arduino core
+  __attribute__((constructor (102)))
+  void HAL_preinit() {
     enableCycleCounter();
-  #endif
+  }
+#endif
 
+// HAL initialization task
+void HAL_init(void) {
```

Comment:
This function declaration isn’t closed. Was it intended to be a function call?

Reply:
The function declaration is closed. The brackets match and my compiler doesn't throw any errors.

Using ```__attribute__((constructor (102)))``` will make the function being called like a ```main()``` function.

Do I maybe understand you incorrect?

Reply:
Maybe the branch just needed a rebase to pass Travis CI… Doing that now.